docs/source/support.rst

```diff
@@ -90,5 +90,5 @@ In addition to the previous options, paid support is available from the
 following organizations (listed in alphabetical order):
 
 -   Anaconda: `<https://www.anaconda.com/products/professional-services>`_
--   Coiled: `<https://coiled.io>`_
+-   Coiled: `<https://coiled.io?utm_source=dask-docs&utm_medium=support>`_
```

Comment:
Actually the link here looks off now 

![Screenshot 2023-05-17 at 4 59 09 PM](https://github.com/dask/dask/assets/11656932/722575a3-11cc-4494-9ce7-afb072e53d80)

Let's just have this list be a company names that are hyperlinked

Reply:
Ah thanks for catching this! Just pushed up a change